tutorials/time-freq/ssvep.py

```diff
@@ -0,0 +1,382 @@
+# Minimal preprocessing
+# ---------------------
+#
+# Due to a generally high SNR in SSVEP/vSSR, typical preprocessing steps
+# are considered optional. this doesnt mean, that a proper cleaning would not
+# increase your signal quality!
+#
+# Raw data comes with FCz recording reference, so we will apply common-average
+# rereferencing.
+
+###############################################################################
+# Set montage
+# ^^^^^^^^^^^
+
+montage_style = 'easycap-M1'
+montage = mne.channels.make_standard_montage(
+    montage_style,
+    head_size=0.095)  # head_size parameter default = 0.095
+raw.set_montage(montage)
+
+###############################################################################
+# Set common average reference
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+raw.set_eeg_reference('average', projection=False)
+
+###############################################################################
+# Apply notch filtering
+# ^^^^^^^^^^^^^^^^^^^^^
+notch = np.arange(raw.info['line_freq'], raw.info['lowpass'] / 2,
+                  raw.info['line_freq'])
+raw.notch_filter(notch, filter_length='auto', phase='zero')
+
+###############################################################################
+# Apply linear filtering
+# ^^^^^^^^^^^^^^^^^^^^^^
+
+hp = .1
+lp = 250.
+raw.filter(hp, lp, fir_design='firwin')
+
+
+###############################################################################
+# Frequency analysis
+# ------------------
+# We use Welch's method for frequency decomposition, since it is really fast.
+# You can compare it with, e.g., multitaper to get an impression of the
+# influence on SNR. All the other methods implemented in MNE can be used as
+# well.
+
+###############################################################################
+# Construct epochs
+# ^^^^^^^^^^^^^^^^
+
+events, _ = mne.events_from_annotations(raw)
+raw.info["events"] = events
+tmin, tmax = -1., 30.  # in s
+baseline = None
+epochs = mne.Epochs(raw, events=events, event_id=event_id['12hz'], tmin=tmin,
+                    tmax=tmax, baseline=baseline)
+
+###############################################################################
+# Calculate power spectral density
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+tmin = 0.
+tmax = 30.
+fmin = 1.
+fmax = 90.
+sf = epochs.info['sfreq']
+
+psds, freqs = mne.time_frequency.psd_welch(
+    epochs,
+    n_fft=int(sf * (tmax - tmin)), n_overlap=int(sf * .5), n_per_seg=None,
+    tmin=tmin, tmax=tmax,
+    fmin=fmin, fmax=fmax)
+
+
+###############################################################################
+# Extract SSVEP/vSSR
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+#
+# The function below calculates the ratio of power in the target frequency bin
+# to average power in a set of neighbor (noise) bins. The composition of noise
+# bins can be tweaked by two parameters:
+# - how many noise bins do you want?
+# - do you want to skip n bins directly next to the target bin?
```

Comment:
those two are not in individual lines/bullets yet..

Reply:
Fixed. It turned out that in restructuredtext the list is marked by asterisks, not dashes. And there needs to be an empty line before lists.